src/Nervos/CellDep.h

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "OutPoint.h"
+#include "../proto/Nervos.pb.h"
+
+namespace TW::Nervos {
+
+enum DepType { DepType_Code = 0, DepType_DepGroup = 1 };
+
+/// Nervos cell dep.
+class CellDep {
+  public:
+    OutPoint outPoint;
+    DepType depType;
+
+    /// Initializes a cell dep with a previous output and depType
+    CellDep(OutPoint outPoint, DepType depType) : outPoint(std::move(outPoint)), depType(depType) {}
+
+    CellDep(const Proto::CellDep& cellDep) : outPoint(cellDep.out_point()) {
+        if (cellDep.dep_type() == "code") {
+            depType = DepType_Code;
+        } else if (cellDep.dep_type() == "dep_group") {
+            depType = DepType_DepGroup;
+        }
```

Comment:
consider writing a function that will set the `depType` field member

Reply:
Done